packages/react-instantsearch-dom/src/core/__tests__/createInstantSearchServer.js

```diff
@@ -601,14 +646,63 @@ describe('createInstantSearchServer', () => {
 
         const [first] = data;
 
+        expect(first._internalIndexId).toBe('index1');
         expect(first.state.index).toBe('index1');
         expect(first.state.query).toBe('iPhone');
 
         const [, second] = data;
 
+        expect(second._internalIndexId).toBe('index2');
         expect(second.state.index).toBe('index2');
         expect(second.state.query).toBe('iPad');
       });
+
+      it('with search state - same index', async () => {
+        const { InstantSearch, findResultsState } = createInstantSearchServer();
+
+        const Connected = createWidget();
+
+        const props = {
+          ...requiredPropsWithSearchClient,
+          indexName: 'index1',
+        };
+
+        const App = ({ searchState }) => (
+          <InstantSearch {...props} searchState={searchState}>
+            <Connected />
+
+            <Index indexId="index1_with_refinement" indexName="index1">
+              <Connected />
+            </Index>
+          </InstantSearch>
+        );
+
+        const data = await findResultsState(App, {
+          searchState: {
+            query: 'iPhone',
+            indices: {
+              // eslint-disable-next-line camelcase
+              index1_with_refinement: {
+                query: 'iPad',
+              },
+            },
+          },
+        });
+
+        expect(data).toHaveLength(2);
+
+        const [first] = data;
+
+        expect(first._internalIndexId).toBe('index1');
+        expect(first.state.index).toBe('index1');
+        expect(first.state.query).toBe('iPhone');
+
+        const [, second] = data;
+
+        expect(second._internalIndexId).toBe('index1_with_refinement');
+        expect(second.state.index).toBe('index1');
+        expect(second.state.query).toBe('iPad');
+      });
     });
   });
```

Comment:
shouldn't we test the behaviour of this too?
```js
     <InstantSearch {...props}>
         <Index indexId="index1" indexName="index1" /> 
         <Index indexId="index1" indexName="index2" />
      </InstantSearch>	      

```

Reply:
At the moment nothing prevents this usage of the API. The latest component mounted on the page will have the "advantage" (we can image the merge like an `Object.assign`).